### ISSUE TITLE
Inventory data.json export error

### ISSUE THREAD

**Issue by FuhuXia** [GSA/data.gov#4237]
There is an error when downloading datajson for [National Renewable Energy Laboratory (NREL)](https://inventory.data.gov/organization/nrel-gov)

In the errorlog.txt
```
<class 'TypeError'> : blueprint.py : 194 : can only concatenate str (not "NoneType") to str
```

Error in the cf logs:
```
INFO  [repoze.who] -- repoze.who request started (/organization/580b3fc1-c5f7-4b3d-8873-81018f782b96/unredacted.json) --
DEBUG [ckanext.datajson.blueprint] Generating JSON for unredacted to 580b3fc1-c5f7-4b3d-8873-81018f782b96 (fuhu_xia_gsa_gov)
DEBUG [ckanext.datajson.package2pod] ZIP ... BECOMES ... application/zip
ERROR [ckanext.datajson.package2pod] <class 'TypeError'> : package2pod.py : 145 : can only concatenate str (not "NoneType") to str
ERROR [ckanext.datajson.package2pod] <class 'TypeError'> : package2pod.py : 145 : can only concatenate str (not "NoneType") to str
ERROR [ckanext.datajson.package2pod] <class 'TypeError'> : package2pod.py : 60 : can only concatenate str (not "NoneType") to str
ERROR [ckanext.datajson.package2pod] <class 'TypeError'> : package2pod.py : 60 : can only concatenate str (not "NoneType") to str
ERROR [ckanext.datajson.blueprint] <class 'TypeError'> : blueprint.py : 194 : can only concatenate str (not "NoneType") to str
ERROR [ckanext.datajson.blueprint] <class 'TypeError'> : blueprint.py : 194 : can only concatenate str (not "NoneType") to str
INFO  [ckan.config.middleware.flask_app]  200 /organization/580b3fc1-c5f7-4b3d-8873-81018f782b96/unredacted.json render time 1.058 seconds
```

## How to reproduce

click Unredacted Inventory button

## Expected behavior

data.json downloaded

## Actual behavior

empty.json and errorlog.txt.

## Sketch
A quick [PR](https://github.com/GSA/ckanext-datajson/pull/138) did not help.
Download prod db and restore in a development environment to replicate the error and debug.

**Comment by Jin-Sun-tts** [comment]
Tried to repeat this issue locally. Could not reproduce this issue locally. But found some new issues:
1. locally instance does not have the ‘Unredacted Inventory’ buttons show up.
2. By testing download from prod and local, noticed most case there were 1 dataset missing in the datajson download file.  

Next will test on staging to see if we may reproduce this problem.

**Comment by Jin-Sun-tts** [comment]
Reproduced this issue on Staging, found the issue was caused by the minetypy of SHP is None, and could not concatenate with string, fixed it by setting it to 'application/octet-stream' when it is None. 

After fixed this issue, found other two other errors like below:

Validation failed, best guess of error:
 'Geotiff' does not match '^[-\\w]+/[-\\w]+(\\.[-\\w]+)*([+][-\\w]+)?$'


Validation failed, best guess of error:
 'Mapping and Assessment of the United States Ocean Wave Energy Resource' is not a 'uri'
 (this one is because user put invalid text in the URL field)

**Comment by Jin-Sun-tts** [comment]
Found the root cause for this issue: By testing download from prod and local, noticed most case there were 1 dataset missing in the datajson download file.

The missing database is 'Draft', so it was not included in the export datajson file as expected. But the confuse part is it shows 'public' in the datasets page.

**Comment by jbrown-xentity** [comment]
In general, the idea of `Drafts` is not well understood in this inventory context. There's some internal flag for the data provider to say "not ready to be released", essentially. I don't remember what the CKAN metadata flag is, but it's custom to what we've built.
All datasets by default are created 'Private', so that no one can see it other than the organization that created it. There is 1 exception: if a file is loaded through CKAN, that file will need to be public. The only way we found for permissions to work, is to make the CKAN dataset 'Public' so that when the dataset is published on catalog, the resource that points back to inventory is accessible. See example on catalog [here](https://catalog.data.gov/dataset/inventory-of-owned-and-leased-properties-iolp) (the resource Download buttons are all inventory URL's). @Jin-Sun-tts
